src/Microsoft.OpenApi.Readers/V2/OpenApiDocumentDeserializer.cs

```diff
@@ -149,39 +149,19 @@ private static void MakeServers(IList<OpenApiServer> servers, ParsingContext con
             {
```

Comment:
If schemes is set to an empty list, I think we should treat that in the same way as schemes being null, i.e. line 148 should be `if (schemes != null && schemes.Count != 0)`

Spec says:
If the schemes is not included, the default scheme to be used is the one used to access the Swagger definition itself.

The term "not included" is not really precisely defined, but I think it should also apply to an empty schemes list. If we don't specifically handle it here, we won't have any `server` when `schemes` is an empty list.